src/nbl/asset/utils/CIESProfileParser.cpp

```diff
@@ -150,12 +157,58 @@ bool CIESProfileParser::parse(CIESProfile& result)
         }
     }
 
-    const double factor = ballastFactor * candelaMultiplier;
-    for (int i = 0; i < hSize; i++)
-        for (int j = 0; j < vSize; j++)
-            result.setValue(i, j, factor * getDouble("intensity value truncated"));
-    
-    result.maxValue = *std::max_element(std::begin(result.data), std::end(result.data));
+    {
+        const double factor = ballastFactor * candelaMultiplier;
+        for (int i = 0; i < hSize; i++)
+            for (int j = 0; j < vSize; j++)
+                result.setCandelaValue(i, j, factor * getDouble("intensity value truncated"));
+    }
+
+    float totalEmissionIntegral = 0.0, nonZeroEmissionDomainSize = 0.0;
+    constexpr auto FULL_SOLID_ANGLE = 4.0f * core::PI<float>();
+
+    if (result.symmetry == CIESProfile::ISOTROPIC && result.vAngles.size() == 1) // not sure if any IES provides this kind of distribution but it should be possible to do so meaning fixed candela value for each direction
+    {
+        const auto candelaValue = result.getCandelaValue(0, 0);
+        constexpr auto solidAngle = FULL_SOLID_ANGLE; // spans whole sphere area
+
+        totalEmissionIntegral = candelaValue * solidAngle;
+        nonZeroEmissionDomainSize = solidAngle;
+    }
+    else
+        for (size_t i = 0; i < result.hAngles.size(); i++)
+            for (size_t j = 0; j < result.vAngles.size(); j++)
+            {
+                const auto candelaValue = result.getCandelaValue(i, j);
+
+                auto getDPhiRad = [&]()
+                {
+                    if (result.symmetry == CIESProfile::ISOTROPIC) // this could be nice templated lambda, TODO: correct once c++20 enters this branch
+                        return core::PI<float>() * 2.0f; // spans whole circle
+                    else
+                        return core::radians<float>((i < result.hAngles.size() - 1) ? result.hAngles[i + 1] - result.hAngles[i] : result.hAngles[i] - result.hAngles[i - 1]);
+                };
+
+                const float thetaRad = core::radians<float>(result.vAngles[j]);
+                const float dThetaRad = core::radians<float>((j < result.vAngles.size() - 1) ? result.vAngles[j + 1] - result.vAngles[j] : result.vAngles[j] - result.vAngles[j - 1]);
```

Comment:
because to integrate you need to compute `cos(loTheta)-cos(hiTheta)` I'd get them individually.

Also because there's no grid wrapparound for this dimension, `j` needs to vary from 0 to `vAngles.size()-1` (one per gap)

Reply:
> Also because there's no grid wrapparound for this dimension, j needs to vary from 0 to vAngles.size()-2 (one per gap)

resolved in following commit